### ISSUE TITLE
next manual 2.11++

### ISSUE THREAD

**Issue by bob-carpenter** [stan-dev/stan#1868]
Updates for next version of manual.

**Comment by bob-carpenter** [comment]
There's a problem on p. 97 with a pointer to `example-models/.../misc/gaussian-process` in the HMM section.

* [ ] remove pointers to example-models repo; include enough detail in manual where necessary

**Comment by chrisranderson** [comment]
Section 1.5, paragraph 2:
![image](https://cloud.githubusercontent.com/assets/5461398/14644584/cec06950-060f-11e6-87ef-1fc3ca2b8d9b.png)

**Comment by chrisranderson** [comment]
Section 1.6, 3rd paragraph:

> If the prior is not uniform, the posterior mode is sometimes
> called the maximum a posterior (MAP) estimate.

I think it's called maximum a posterior**i**.

**Comment by chrisranderson** [comment]
Section 3.2, upper and lower bounds:

> Variables may be declared with constraints All of the basic data types may be given
lower and upper bounds using syntax such as

Missing punctuation before "All". Picky suggestion: could be good to use a colon after "as".

**Comment by daydreamt** [comment]
Section 29.1:

I think that the int_literal in the stan grammar (which appears only once and is not defined) refers to integer_literal (which is defined below).



![int to integer](https://cloud.githubusercontent.com/assets/903245/14761946/90ddbc16-096e-11e6-9c91-2b892c221561.png)

**Comment by skanskan** [comment]
Where can it be downloaded?
Thanks

**Comment by skanskan** [comment]
OK, I've downloaded and compiled it with "make".
Can we also download the stan 3.0 manual?

**Comment by bob-carpenter** [comment]
GitHub issues aren't for general questions, so I'm going to remove this comment and the above comments.  Please use the stan-users mailing list for this kind of thing.

There's no Stan 3 manual because there's no Stan 3 yet!

**Comment by bob-carpenter** [comment]
Mike Lawrence on stan-users suggested:

* adding a description of why `if_else` is more efficient than multiplying by 0/1

In general, I wrote this back on the list, and it should be part of the intro to the optimization chapter 

* [ ] add "efficiency" to the title

* [ ] add discussion of basic principles to intro

Efficiency involves two issues in Stan, and both stem from first principles:

1.  Statistical Efficiency.  This is all about posterior curvature in Stan.  Some posteriors are easy to move around than others and changing parameterizations can help.  It can also help to marginalize and compute sufficient statistics where possible.  (This is also  true of Gibbs and Metropolis, by the way.)

2.  Computational Efficiency.  This is all about efficient derivatives, and the smaller the resulting expression graph, the faster the derivatives.  That's why vectorization involving parameters gives you an efficiency boost but vectorization of constants doesn't.  

The `if_else` comparison to multiplying by 0 or 1 stems from point 2 --- there are fewer terms in the expression graph for the `if_else`.   But there's a catch. The `if_else` evaluates both sides, so in general, a conditional is required, so instead of 

```
  a <- a + if_else(cond, b, c)
```

where `b` and `c` are complex expressions, it's more efficient to do this

```
  if (cond)
    a <- a + b;
  else
    a <- a + c;
```

We're adding a ternary operator `(cond ? a : b)` that'll avoid the eager evaluation of the RHS term in the `if_else` and also work for more general types.

**Comment by bob-carpenter** [comment]
* [ ] add discussion from stan-users on why even simple reparameterizations like centered vs. non-centered affect max likelihood but not Bayes

```
Surprising (the result, not that Michael's right about calc!).
It looks like we're optimizing f in one case and h in the second,
where

 f(y_raw, mu, sigma,theta) = g(sigma * y_raw + mu, mu, sigma, theta)
                              + log normal(y_raw | 0, 1)

vs.

 h(y, mu, sigma, theta) = g(y, mu, sigma, theta) 
                          + log normal(y | mu, sigma)

                        = g(y, mu, sigma, theta)
                          + log normal((y - mu)/sigma | 0, 1)
                          - log(sigma)

where g is just the rest of the model log density.
The only difference is the -log(sigma) term, but that will
cause different densities.
```

**Comment by bob-carpenter** [comment]
* [ ] add chapter detailing deprecations and what to replace them with

As is, there is no doc for any deprecated functions.

**Comment by bob-carpenter** [comment]
@infotroph (Chris Black) commented that `programming.tex` has a `s igma` typo around line 2000.

* [ ] fix it

Original comment on an earlier pull request:  https://github.com/stan-dev/stan/commit/462b128f327762295638915167d5ba7c8b76b1c9#commitcomment-18006886

**Comment by bob-carpenter** [comment]
From @andrewgelman:

possible ambiguity in manual:

On page 412:

```
n ~ poisson(lambda);
```
Increment log probability with `poisson_log(n, lambda)`, dropping constant additive terms; Section 27.3 explains sampling statements. 

Then on page 413:

```
n ~ poisson_log(alpha);
```
Increment log probability with `poisson_log_log(n, alpha)`, dropping con- stant additive terms; Section 27.3 explains sampling statements.

--

It seems to me that the poisson_log(n,lambda) function on page 412 is not the same as the poison_log(alpha) function on page 413 <i>[comment:  technically the former is not a function, just part of that compound `~` statement]</i>. I recognize that this is not a strict ambiguity as the two functions have different numbers of arguments, but it could be confusing.  Perhaps would be less ambiguous to write as follows:

On page 412:

```
n ~ poisson(lambda);
```
Increment log probability with `poisson_lpmf(n, lambda)`, dropping constant additive terms; Section 27.3 explains sampling statements. 

Then on page 413:

```
n ~ poisson_log(alpha);
```
Increment log probability with `poisson_log_lpmf(n, alpha)`, dropping con- stant additive terms; Section 27.3 explains sampling statements.

<i>[ed.  What needs to happen is that the macro expanding the sampling statements needs to be changed exactly the way Andrew suggests.]</i>

**Comment by bob-carpenter** [comment]
From @sskates2 as issue comment https://github.com/stan-dev/stan/issues/1811#issuecomment-233791652 :

There are multiple places in the Stan Manual 2.10 where `dist_lpdf(y | theta1, ..., thetaN)` is written as `dist_lpdf(y, theta1, ..., thetaN)` - i.e. without the vertical bar. The "comma" version results in a parser error.

**Comment by bob-carpenter** [comment]
* [ ] Write Stan functions for the built-in transforms and Jacobians, or expose the ones that are already in the math library (that'd be a different issue).

**Comment by bob-carpenter** [comment]
* [ ] add parens around `(i==j ? 0.1 : 0.0)` in GP example
* [ ] thank Wiktor Soral for pointing this out on the users list

**Comment by bob-carpenter** [comment]
From @andrewgelman:

* [ ] fix: In bottom of p.125, "[5]" should be "[N]".

**Comment by syclik** [comment]
From @wrobell
- [ ] insert missing `(` in change point model: `lp[s] = lp[s] + poisson_lpmf(D[t] | t < s ? e : l));` (before the `t < s`). #1998

**Comment by jmegner** [comment]
There are 2 typos related to Figure 26.3 (page 280) in the reference manual.  The third entry in the table exemplifying 2 single-valued indices should be "a[i, j]" instead of "a[is, js]".  Also, the caption should have the word "argument" instead of "argumnt".

I have forked and made a [commit](https://github.com/jmegner/stan/commit/8422585e416d9509ee1743f9a1ac50aae352c9de) with a change to fix these two typos.  I even have prepared a pull request, but thought that it might be better for me to comment here.

Any corrections to my reporting/submitting this typo fix is appreciated.  Thank you for your wonderful work on stan.

**Comment by bob-carpenter** [comment]
Thanks.  If you turn that commit into a pull request, I can merge
into develop (we'll leave this comment to be sure to thank you
in our manual acknowledgements section).

- Bob

**Comment by bob-carpenter** [comment]
Or feel free to leave it and I'll take care of it when I update
the manual for 2.12.0 (we better get to 3.0.0 before 2.100.0!).

**Comment by bob-carpenter** [comment]
* [ ] remove comment/footnote suggesting we'll add discrete parameters in Stan 2, as we're well into Stan 2's lifecycle and have no plans to add discrete parameters any time soon

**Comment by bob-carpenter** [comment]
* [ ] thank Stephen Hoover for code patches

**Comment by bob-carpenter** [comment]
* [ ] switch all lpdf, lcdf, and lccdf function signatures to use `|` rather than `,`

**Comment by juhokokkala** [comment]
Section 42.3. Negative Binomial Distribution (log alternative parametrization) (\label{neg-binom-2-log.section}) claims that the sampling statement is
 `y ~ neg_binomial_2_lpmf(eta, phi);`
I believe this should be 
 `y ~ neg_binomial_2_log(eta, phi);`
as that's how it was in 2.9, and when compiling with the "new sampling statement," I get an error, the  message includes 
`c:/temp/negbinmodel.hpp:165:32: error: 'neg_binomial_2_lpmf' was not declared in this scope `.
When using  `y ~ neg_binomial_2_log(eta, phi);`, the model compiles just fine.

Based on a quick look, this originates from commit df3fd72 -- possibly a find-replace error as the intent may have been to replace the log target increment function of the alternative parametrization (no log)? 

Can someone please confirm that `neg_binomial_2_log` is still the correct sampling statement in 2.11? (I wonder if I should have posted as a separate issue instead of this manual thread?)

**Comment by bob-carpenter** [comment]
@juhokokkala  Thanks for noticing this.  Yes, this is a doc issue and should go here.  You're correct that the sampling statements haven't changed.  I'll fix the doc for the next release.

**Comment by juhokokkala** [comment]
Ok, thanks for the quick reply.

**Comment by bob-carpenter** [comment]
Emmanuel Charpentier suggested via email that the print form would be more pleasant with proper chapter and part typesetting, and he sent along a simple patch.

```
-\documentclass[10pt,openany]{book}
+\documentclass[10pt,twoside,openright]{book}
```

* [ ] patch the doc class
* [ ] thank Emmanuel

**Comment by bob-carpenter** [comment]
* [x] 2.12 version update
